compiler/src/dotty/tools/dotc/transform/init/Semantic.scala

```diff
@@ -424,21 +471,36 @@ object Semantic:
     def report(err: Error): Unit
     def reportAll(errs: Seq[Error]): Unit = for err <- errs do report(err)
 
+  /** A TryReporter cannot be simply thrown away
+   *
+   *  Either `abort` should be called or the errors be reported.
+   */
+  trait TryReporter extends Reporter:
```

Comment:
Add comment to say what `abort` does. Does it reset the caches to their state before the reporter was created?

Reply:
Yes, it reverts the cache to the previous state. I'll add a comment.